tasty-query/shared/src/main/scala/tastyquery/Types.scala

```diff
@@ -768,17 +771,19 @@ object Types {
   class NonMethodReference(val kind: String) extends Exception(s"reference to non method type in $kind")
 
   /** The singleton type for path prefix#myDesignator. */
-  case class TermRef(override val prefix: Type, var myDesignator: Designator)
+  final class TermRef private (val prefix: Type, myDesignator: TermSymbol | TermName | LookupIn | Scala2ExternalSymRef)
       extends NamedType
       with SingletonType
       with ImplicitRef {
 
     type ThisName = TermName
     type ThisSymbolType = TermSymbol
+    protected[this] type ThisDesignatorType = TermSymbol | TermName | LookupIn | Scala2ExternalSymRef
 
-    override def designator: Designator = myDesignator
+    override def designator: ThisDesignatorType = myDesignator
```

Comment:
this `designator` is public, is this intentional?

Reply:
Ouch! No, definitely not intentional.